apps/chat/src/components/IframeRenderer/index.tsx

```diff
@@ -0,0 +1,143 @@
+import React, {
+  Ref,
+  forwardRef,
+  useCallback,
+  useEffect,
+  useImperativeHandle,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+
+import { useRouter } from 'next/router';
+
+import { Spinner } from '../Common/Spinner';
+
+import {
+  AttachmentData,
+  VisualizerConnectorEvents,
+  VisualizerConnectorRequest,
+  VisualizerConnectorRequests,
+} from '@epam/ai-dial-shared';
+import { VisualizerConnector } from '@epam/ai-dial-visualizer-connector';
+
+interface IframeRendererProps {
+  iframeUrl: string;
+  title: string;
+  width?: number | string;
+  height?: number | string;
+  onMessage?: (event: MessageEvent) => void;
+  containerStyle?: React.CSSProperties;
+  containerClassName?: string;
+  conversationId?: string;
+}
+
+export const IframeRenderer = forwardRef<HTMLDivElement, IframeRendererProps>(
+  (
+    {
+      iframeUrl,
+      title,
+      width = '100%',
+      height = '100%',
+      onMessage,
+      containerStyle = {},
+      containerClassName = '',
+      conversationId,
+    },
+    ref: Ref<HTMLDivElement>,
+  ) => {
+    const containerRef = useRef<HTMLDivElement>(null);
+    const visualizer = useRef<VisualizerConnector | null>(null);
+
+    const router = useRouter();
+
+    const isPreviewConversation = useMemo(() => {
+      return router.pathname === '/apps-editor/[slug]/settings';
+    }, [router.pathname]);
+
+    const [loading, setLoading] = useState<boolean>(true);
+
+    useImperativeHandle(ref, () => containerRef.current as HTMLDivElement);
+
+    useEffect(() => {
+      if (containerRef.current && !visualizer.current) {
+        visualizer.current = new VisualizerConnector(containerRef.current, {
+          domain: iframeUrl,
+          hostDomain: window.location.origin,
+          visualizerName: title,
+        });
```

Comment:
Should we destroy if props have changed to avoid memory leak?
```suggestion
        if (visualizer.current) {
          visualizer.current.destroy();
        }
    
        visualizer.current = new VisualizerConnector(containerRef.current, {
          domain: iframeUrl,
          hostDomain: window.location.origin,
          visualizerName: title,
        });
```

Reply:
done